src/components/LogoutModal.js

```diff
@@ -0,0 +1,143 @@
+import React, { useState, useContext } from 'react';
+import { Alert, Modal, StyleSheet, Text, Pressable, View } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
+
+import { BiteShareContext, biteShareState } from '../BiteShareContext';
+import { colors } from '../infrastructure/colors';
+import { signOutUser } from '../../firebase/helpers/authentication.firebase';
+import { deleteADocument, getADocReferenceFromCollection, updateADocument, getAllDocuments } from '../../firebase/helpers/database.firebase';
+
+const LogoutModal = ({ modalVisible, setModalVisible }) => {
+  const { state: { guests, accountType, sessionId, nickname }, dispatch } = useContext(BiteShareContext);
+  const navigation = useNavigation();
+  const styles = StyleSheet.create({
```

Comment:
I would suggest move the style definition out of the component? otherwise it might be rerun every time?

Reply:
Sorry, I'm not sure exactly what you mean?

Reply:
Oh I meant move `const styles = StyleSheet.create({...})` outside of `LogoutModal`